### ISSUE TITLE
[FEATURE REQUEST] Get the list of available extensions directly

### ISSUE THREAD

**Issue by dsmtE** [OpenImageIO/oiio#2615]
Hello,
We need to know the extensions supported by your library.
Until now we used the **oiio.get_string_attribute ("extension_list")** function which returns a std::string. But the problem is that it has to be parsed by us to be used effectively.

Is there a function to get a variable of type std::vector<std::string>, std::map<std::string, std::vector<std::string>> or things like that to directly get the list of usable extensions?

_You can find the definition of our function through this commit https://github.com/alicevision/AliceVision/pull/806/commits/ab3c9e3dbb69426268a8563cb8028d26127bcc04.
Or directly here  https://github.com/alicevision/AliceVision/blob/dev/imageProcessing/src/aliceVision/image/io.cpp._

Could you integrate such a function allowing us to use it directly without having to parse it by ourselves ?

**Comment by DougRogers** [comment]
The second link is broken.

**Comment by lgritz** [comment]
I think this should work and is fairly compact:

```
std::map<std::string, std::vector<std::string>> extension_map;  // <-- the result you want
std::string all_extensions = OIIO::get_string_attribute("extension_list");
for (auto oneformat : OIIO::Strutil::splitsv(all_extensions, ";")) {
    auto format_exts = OIIO::Strutil::splitsv(oneformat, ":", 2);
    if (format_exts.size() != 2)
        continue;   // something went wrong
    extension_map[format_exts[0]] = OIIO::Strutil::splits(format_exts[1], ",");
}
```

**Comment by lgritz** [comment]
But your request is reasonable, I'll encapsulate that in top-level function for convenience.

**Comment by lgritz** [comment]
Proposed fix in #2904
